### PR TITLE
Added cachetools to pyproject.toml

### PR DESCRIPTION
**Pull request by annahedstroem**
### Description
- Tiny PR fixing Python package issue: https://github.com/understandable-machine-intelligence-lab/Quantus/actions/runs/8273585187/job/22637606634 

### Minimum acceptance criteria
- Specify what is necessary for the PR to be merged with the main branch.
- @mentions of the person that is apt to review these changes e.g., @annahedstroem

### CODE REVIEW

**Comment by aaarrti** [comment]
Hi @annahedstroem, e.g. in [this](https://github.com/understandable-machine-intelligence-lab/Quantus/actions/runs/8281420255/job/22659852897) run if you check output of `tox install`, you could see:

- py38: keras==2.13.1
- py39: keras==3.0.5

image

image

And the actuall exception comes from Keras V3 trying to read model weights 

```
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
  tests/conftest.py:48: in load_cifar10_model_tf
      model.load_weights("tests/assets/cifar_tf_weights.keras")
  .tox/py39/lib/python3.9/site-packages/keras/src/utils/traceback_utils.py:123: in error_handler
      raise e.with_traceback(filtered_tb) from None
  /opt/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/zipfile.py:1266: in __init__
      self._RealGetContents()
  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
```
From [here](https://keras.io/guides/serialization_and_saving/)
image

we could get a hint, that they do not support old serialisation format anymore.

### To sum up 
1. As hotfix pin the version in
```toml
[project.optional-dependecies]
tensorflow = ["keras<3", ...]
```
2. Let's think if you we would want to upgrade to KerasV3 (there will be a lot to do)
